twitter/twitter.php

```diff
@@ -173,6 +174,39 @@ function twitter_follow(App $a, array &$contact)
 	}
 }
 
+function twitter_unfollow(App $a, array &$hook_data)
+{
+	$contact = $hook_data['contact'];
+	Logger::info('Check if contact is twitter contact', ['url' => $contact["url"]]);
+
+	if (!strstr($contact["url"], "://twitter.com") && !strstr($contact["url"], "@twitter.com")) {
+		return;
+	}
+
+	// contact seems to be a twitter contact, so continue
+	$nickname = preg_replace("=https?://twitter.com/(.*)=ism", "$1", $contact["url"]);
+	$nickname = str_replace("@twitter.com", "", $nickname);
+
+	$uid = $a->getLoggedInUserId();
+
+	$ckey = DI::config()->get('twitter', 'consumerkey');
+	$csecret = DI::config()->get('twitter', 'consumersecret');
+	$otoken = DI::pConfig()->get($uid, 'twitter', 'oauthtoken');
+	$osecret = DI::pConfig()->get($uid, 'twitter', 'oauthsecret');
+
+	// If the addon is not configured (general or for this user) quit here
+	if (empty($ckey) || empty($csecret) || empty($otoken) || empty($osecret)) {
+		return;
+	}
+
+	try {
+		$connection = new TwitterOAuth($ckey, $csecret, $otoken, $osecret);
+		$connection->post('friendships/destroy', ['screen_name' => $nickname]);
```

Comment:
I guess this `post` returns some value. It would be nice to have the result logged in the `info` level as well.

Reply:
Of course.